### ISSUE TITLE
How to format code in <script> tags?

### ISSUE THREAD

**Issue by trusktr** [prettier/prettier#2648]
I've got `.hbs` files with `<script>` tags in them. How can I format those? I'm currently using [eslint-plugin-script-tags](https://github.com/mapbox/eslint-plugin-script-tags) to lint `<script>` tags in there. Maybe I can use eslint-plugin-prettier with this?

If not, is there another way other than hacking with `range-start` and `range-end`?

**Comment by azz** [comment]
Check out https://github.com/prettier/prettier/issues/2097#issuecomment-321435696 (and that issue I'm general). It's a planned feature but not supported just yet.

**Comment by trusktr** [comment]
I saw that issue. Turns out it is simple in that linked article to pass source to prettier using html-script-hook.

But what I'm going to do, which I think is more semantic and idiomatic is to just move all JS into actual `.js` files. Then I'll modify my build system accordingly. Prettier will just work that way.

Even with `Vue`, it will automatically "import" JS files from `<script src="...">`, and stick them into the resulting JS bundle. So doing that will simply work as well, without you having to fix 2097.

Actually, eslint has this feature coming soon, a PR: https://github.com/eslint/eslint/pull/9090

This allows processor plugins to work with "fix" mode.

What this means is that we can use `eslint-plugin-script-tags` and `eslint-plugin-prettier` along with the eslint `--fix` option, and it will simply work. It works with .hbs, .php, .html files, etc, any file containing `<script>` tags. It's just not released yet.
